package.json

```
@@ -1,161 +1,9 @@
 {
-  "name": "ember-file-upload",
-  "version": "5.0.0-beta.3",
-  "description": "An uploader component for Ember apps",
-  "keywords": [
-    "ember-addon",
-    "file-upload"
+  "private": "true",
+  "workspaces": [
+    "ember-file-upload"
   ],
```

Did you considered having the workspaces in a folder called `packages/` instead of the repository root? That's a pattern at least I have seen often. It also allows to use a wildcard:

```suggestion
  "workspaces": [
    "packages/*"
  ],
```

That folder could contain later `ember-file-upload` itself, the docs app and libraries like `ember-file-upload-mirage-handler`.

When migrating to v2 addon format, we could add a `test-packages/` or `test-apps/` folder and move the dummy app to that one.

---

Yeah, there are no established practices around that, how ember addons in v2 are organized on a repo-level. What I can see in the wild:

1. `packages/*` with [ember-cite](https://github.com/gossi/ember-cite) or [ember-stargate](https://github.com/kaliber5/ember-stargate/tree/master/packages) (they could be at root level)
2. `$packageName` and `testing/*` with [ember-resources](https://github.com/NullVoxPopuli/ember-resources)
3. Or not a monorepo as mentioned here: https://www.kaliber5.de/de/blog/v2-addon_en

I'd personally leaning more towards the second option (maybe because I tried the first one?). Now this is becoming also more interesting with (one) more package(s) in this repo, but tests living maybe in the same testing-package? This is a new challenge with this project. In the end, doesn't matter too much I'd say. As that can be organized back and forth to find the best fit 👍